msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -220,6 +220,12 @@ public void signIn(@NonNull final Activity activity,
                        @NonNull final String loginHint,
                        @NonNull final String[] scopes,
                        @NonNull final AuthenticationCallback callback) {
+        final IAccount persistedAccount = getPersistedCurrentAccount();
```

Comment:
nit : Consider using a helper function for this function and call everywhere.

Reply:
This is tricky, because what we're doing with persistedAccount object are different case-by-case.